src/lambda/commands/uploadToAwsCommand.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { createQuickPick, DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { createCommonButtons } from '../../shared/ui/buttons'
+import { uploadLambdaCommand } from './uploadLambda'
+import { CancellationError } from '../../shared/utilities/timeoutUtils'
+import { uploadFileCommand } from '../../s3/commands/uploadFile'
+import { createRegionPrompter } from '../../shared/ui/common/region'
+import { Wizard } from '../../shared/wizards/wizard'
+import { DefaultS3Client } from '../../shared/clients/s3Client'
+
+export interface UploadToAwsWizardState {
+    readonly resource: 's3' | 'lambda'
+    readonly region?: string
+}
+
+function createChooseResourcePrompter() {
+    const items: DataQuickPickItem<'s3' | 'lambda'>[] = [
+        { label: 'Upload to S3', data: 's3', detail: undefined },
+        { label: 'Upload to Lambda', data: 'lambda', detail: undefined },
+    ]
+    return createQuickPick(items, {
+        title: 'Choose an action',
+        buttons: createCommonButtons(),
+    })
+}
+
+export class UploadToAwsWizard extends Wizard<UploadToAwsWizardState> {
+    constructor() {
+        super()
+        this.form.resource.bindPrompter(() => createChooseResourcePrompter())
+        // region is needed to create an S3 client for the upload command
+        this.form.region.bindPrompter(() => createRegionPrompter().transform(region => region.id), {
+            showWhen: ({ resource }) => resource === 's3',
```

Comment:
Despite the comment, I don't understand why we are creating a region prompter here. 

I would expect the "upload to s3" logic here: https://github.com/aws/aws-toolkit-vscode/blob/b20e150914a817600d2ca416b6623b1bb5440443/src/s3/activation.ts#L59-L72

to be extracted to a function, and that function would be called if "Upload to S3" is chosen from this wizard.

The idea is that "Upload to AWS" is a single menu that dispatches to various wizards. We don't want to duplicate parts of the various wizards here.

Reply:
This logic forces them to use a default region, which is what I was originally avoiding, but if no one is complaining about the current logic then I'll go ahead and refactor this a little.

Reply:
> This logic forces them to use a default region

Could add a "switch region" item to the menu similar to: https://github.com/aws/aws-toolkit-vscode/blob/2cfed26f0701afc0230ce39246772aeeeb6c6334/src/shared/ui/common/regionSubmenu.ts#L43

The default region is usually what is wanted. And the "Switch region" item gives an escape hatch for other cases. And avoids an extra menu.